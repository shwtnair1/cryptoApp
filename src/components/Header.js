import React,{useState} from 'react';
import Coin from './Coin';

const Header = (props) =>{
    let [search,setSearch] = useState('');
    

    function handleChange(e){
        e.preventDefault();
        setSearch(e.target.value);
    }

    const filteredCoins = props.coins.filter((coin)=> coin.name.toLowerCase().includes(search.toLowerCase()))
    return (
        <div className='appcontainer'>
        <div className='searchcontainer'>
            <form>
                <input onChange={handleChange} placeholder='Provide coin name' value={search}/>
            </form>
        </div>
        {filteredCoins.map((coin,i)=>{
            return <Coin 
                    key={i} 
                    name={coin.name} 
                    image={coin.image}
                    symbol={coin.symbol} 
                    marketcap={coin.market_cap}
                    price={coin.current_price}
                    pricechange={coin.price_change_24h}
                    />
        })}
        </div>
    )
}

export default Header;