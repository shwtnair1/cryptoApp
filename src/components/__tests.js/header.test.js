import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../Header';
Enzyme.configure({ adapter: new Adapter() });

const coin1 = [{
    "image":"https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
    "name":"Bitcoin",
    "price":3106272,
    "pricechange":3.44971,
    "marketcap":1990031721822
}];
const coin2=[];

describe('Header renders',()=>{
    const instance = shallow(<Header />);

    it('should render correctly',()=>{
        expect(instance).toMatchSnapshot();
    })

    it('should have props defined',()=>{
        instance.setProps({coins:coin1})
        expect(instance.props()).toBeDefined();
    })

    it('should render one form',()=>{
        expect(instance.find('form')).toHaveLength(1)
    })

    it('should render one input',()=>{
        expect(instance.find('input')).toHaveLength(1)
    })

    it('should render Coin component when coin prop is present',()=>{
        instance.setProps({coins:coin1})
        expect(instance.find('Coin')).toHaveLength(1)
    })
    it('should render Coin component when coin prop is empty',()=>{
        instance.setProps({coins:coin2})
        expect(instance.find('Coin')).toHaveLength(0)
    })
})

describe('Header form interactions',()=>{
    const instance = shallow(<Header />);

    it('it should change input value ,when onChange is invoked',()=>{
        instance.find('input').simulate('change',{target:{value:'bit'}})
        expect(instance.find('input').prop('value')).toBe('bit');
    })
})