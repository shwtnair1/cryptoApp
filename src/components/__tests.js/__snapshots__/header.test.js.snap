// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders should render correctly 1`] = `ShallowWrapper {}`;
